### ISSUE TITLE
[Feature Request] Save to post meta with block data

### ISSUE THREAD

**Issue by hj-collab** [nk-crew/lazy-blocks#244]
I had raised a request regarding this before but I was not clear about it. I am okay if this feature request comes in the upcoming paid version. I had written my own custom block as I wanted to save data into post_meta as well as block attributres. I feel it will not be difficult to add. This is how I do it.

I write to attributes as well as meta on onChange. Check screenshot:-
![image](https://user-images.githubusercontent.com/101234934/173226953-6bc43f4c-ef44-4be2-9570-e7b1ca83c5d8.png)

Reason is that If I only read and save to meta then I won't enjoy auto save, post revisions and ability to use full editor area and not just small size sidebar panel. Check screenshot:-

![unknown](https://user-images.githubusercontent.com/101234934/173227013-131b3226-4109-44c6-a9f8-59b53033347c.png)

@nk-o Please take a look and include this in your paid version.

**Comment by nk-o** [comment]
Hey.

I don't quite understand the reason. Currently data saved in the Meta automatically by Gutenberg, and you can output this attribute using function `get_lzb_meta`. It will be displayed both in editor and in frontend.

We don't need to duplicate data inside the block, you just work with Meta directly.

**Comment by hj-collab** [comment]
I will make a video and explain.

**Comment by davidwebca** [comment]
Sorry to interject. @hj-collab do you mean that you want the attribute (such as SKU in your example) to save to the post_meta of the currently post being currently edited by that block?

**Comment by davidwebca** [comment]
Oh I totally understand. It probably doesn't undo properly and doesn't save post_revisions properly because there's no markup changing in the block at all. 

A simple solution would be to add a "revision" attribute that would simply be filled with the unix timestamp at the same rate as a normal block attributes save cycle. It would trigger post revisions, but there would still be problems with undo. 

The other solution, as you said, would be to save to both block attributes and post meta, but prioritize post meta if it's found... though that would run the risk of creating conflicts with other blocks unless there's proper block-to-meta mapping - or left willingly naïve and leave the responsibility to the developer to manage this.

**Comment by hj-collab** [comment]
@davidwebca Also autosave. Autosave is important when you are working with lots of field and lots of data to enter. 

It should read from block data only else autosave, post revisions and undo won't work and write to block data and meta. I need meta because I have areas where getting meta data is easier then parsing block data. This also gives easy way to move out of block in case you face some limitations and can use the meta data with anything else.

**Comment by nk-o** [comment]
I have just changed the deprecated meta attributes and used `useEntityProp` directly, but it does not change anything, it works in the same way.

I have googled and looks like WordPress doesn't autosave meta at all, it is by design. If you have any ideas on how to improve the autosave function properly, I want to look at a solution.

Saving both block attribute and meta is a bad idea, because:
1. Data duplication
2. Autosaveв post will save the block attribute, but not the meta, and we will have outdated meta in autosaved posts, which is not good too
